From 34a7ec821b11f7bed3dd644bf341cbaf6023516b Mon Sep 17 00:00:00 2001
From: Qin Su <qsu@ti.com>
Date: Tue, 11 Sep 2018 17:11:07 -0400
Subject: [PATCH] enable use of boost shared library

Upstream-Status: Inappropriate [configuration]
This is a hack to modify hard coded configuration. If anything, it
should be made configurable.

Signed-off-by: Qin Su <qsu@ti.com>
---
 cmake/GlobalConfig.cmake | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)
 mode change 100644 => 100755 cmake/GlobalConfig.cmake

diff --git a/cmake/GlobalConfig.cmake b/cmake/GlobalConfig.cmake
old mode 100644
new mode 100755
index 2dbeada..a5a1113
--- a/cmake/GlobalConfig.cmake
+++ b/cmake/GlobalConfig.cmake
@@ -96,8 +96,10 @@ endif()
 set(CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cmake/modules ${CMAKE_MODULE_PATH})

 # Boost
-add_definitions("-DBOOST_ALL_NO_LIB") # Turn off auto-linking as we specify the libs manually
-set(Boost_USE_STATIC_LIBS ON)
+#add_definitions("-DBOOST_ALL_NO_LIB") # Turn off auto-linking as we specify the libs manually
+#set(Boost_USE_STATIC_LIBS ON)
+add_definitions(-DBOOST_ALL_DYN_LINK) # use shared library
+set(Boost_USE_STATIC_LIBS OFF)
 find_package(Boost 1.59 REQUIRED COMPONENTS unit_test_framework system filesystem log program_options)
 include_directories(SYSTEM "${Boost_INCLUDE_DIR}")
 link_directories(${Boost_LIBRARY_DIR})
--
1.9.1
